### PR TITLE
Simplify `useCommentCount` server logic

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Refactors `useCommentCount` to make a single decision about `isServer`

Remove the conditional call to `refreshInterval`, as it does not apply on the server.

## Why?

Couple decisions where they matter.

Improvement on #8623 

## Screenshots

N/A

### CODE REVIEW

File: dotcom-rendering/src/lib/useCommentCount.ts [hunk 1]
```diff
@@ -4,32 +4,35 @@ import { useApi } from './useApi';
 
 type CommentCounts = Record<string, number>;
 
-const getInitialSetFromDOMAttribute = (attribute: string) =>
-	[...document.querySelectorAll(`[${attribute}]`)]
-		.map((element) => element.getAttribute(attribute))
-		.filter(isNonNullable);
+const DATA_ATTRIBUTE = 'data-discussion-id';
```
Comment (jamesgorrie) [inline dotcom-rendering/src/lib/useCommentCount.ts:7]: I've always wondered what this pattern is for? My worry is that it's previously allowed someone to export it, and then couple their modules (I can't remember where I've seen this) and why not just use the string?

File: dotcom-rendering/src/lib/useCommentCount.ts [hunk 2]
```diff
@@ -4,32 +4,35 @@ import { useApi } from './useApi';
...
+	uniqueDiscussionIds?.add(shortUrl);
 
-	const searchParams = new URLSearchParams({
-		'short-urls': [...uniqueDiscussionIds]
-			.sort() // ensures identical sets produce the same query parameter
-			.join(','),
-	});
-	const url = `${discussionApiUrl}/getCommentCounts?${searchParams.toString()}`;
-	const { data } = useApi<CommentCounts>(url, {
+	const { data } = useApi<CommentCounts>(getUrl(discussionApiUrl), {
```
Comment (jamesgorrie) [inline dotcom-rendering/src/lib/useCommentCount.ts:33]: I think it's clear, but would leaving a note saying that `uniqueDiscussionIds` only gets generated on the client, and thus this only runs on the client, outlining the issue we've just had might avoid a good citizen trying to simplify later falling into the same trap?

I found [this link](https://swr.vercel.app/docs/conditional-fetching) useful too as it's a bit of a magic API. `null | undefined` means ignore.

Comment (jamesgorrie) [reply dotcom-rendering/src/lib/useCommentCount.ts:33]: Not sure if just inlining this method would be easier to read, but would mean we'd have to align [`useApi`'s `url` property](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/lib/useApi.tsx#L34) to resemble [SWR's `Key` type](https://github.com/vercel/swr/blob/main/_internal/src/types.ts#L306)

**Review by jamesgorrie — APPROVED** [review]

File: dotcom-rendering/src/lib/useCommentCount.ts (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/lib/useCommentCount.ts:7]: What about [more coupling](https://github.com/guardian/dotcom-rendering/pull/8628)? This actually makes it very explicit that there is a dependency between these two values.

Planning to revisit how to improve this overall pattern in #8616

File: dotcom-rendering/src/lib/useCommentCount.ts (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/lib/useCommentCount.ts]: I’ve updated the code to add more documentation and make the method more decriptive: `maybeGetUrl`.

**Pull request merged**